src/predictor/array_tree_layout.h

```diff
@@ -0,0 +1,273 @@
+/**
+ * Copyright 2021-2025, XGBoost Contributors
+ * \file array_tree_layout.cc
+ * \brief Implementation of array tree layout -- a powerfull inference optimization method.
+ */
+#ifndef XGBOOST_PREDICTOR_ARRAY_TREE_LAYOUT_H_
+#define XGBOOST_PREDICTOR_ARRAY_TREE_LAYOUT_H_
+
+#include <limits>
+#include <vector>
+
+namespace xgboost::predictor {
+
+/**
+ * @brief The class holds the array-based representation of the top levels of a single tree.
+ * 
+ * \tparam TreeType The type of the origianl tree (RegTree or MultiTargetTree)
```

Comment:
Strange mix of `@brief` and `\tparam`.
It would be more coherent to use either (`@brief` and `@tparam`) or (`\brief` and `\tparam`)